travel_platform/smart_ordering.py

```diff
@@ -0,0 +1,42 @@
+data = [{'name': 'Pavlo', 'age': 27, 'gender': 'male'},
+        {'name': 'Kate', 'age': 28, 'gender': 'female'},
+        {'name': 'Duc', 'age': 48, 'gender': 'male'},
+        {'name': 'Duc1', 'age': 48, 'gender': 'male'},
+        {'name': 'Duc', 'age': 45, 'gender': 'male'},
+        {'name': 'Sasha'},
+        {'name': 'Sasha1'},
+        {'name': 'Sasha2'},
+        {'age': 60}, {'age': 10}, {'age': 20},
+        {'name': 'Roman', 'gender': 'male'}]
+
+def smart_ordering(data_dict, filter_by, order_by='ASC', limit=100):
+    '''
+    Sorting data by different conditions.
+    Args:
+        data (list): List of dictionaries with data to sort.
+        filter_by (str): Filtering attribute.
+        order_by (str:ASC|DESC): Filtering order.
+        limit (int): Limitation of data to return.
+
+    Returns:
+        list: Sorted list of dictionaries with applied filters.
+    '''
+    if order_by == 'DESC':
+        sorted_data = sorted(data_dict, key=lambda note: note.get(filter_by, 'z'), reverse=True)
```

Comment:
If wanted key is missing, letter "z" sends dictionary to the end of list as unnecessary.